frontend/discourse/app/models/tag.js

```diff
@@ -10,14 +10,45 @@ export default class Tag extends RestModel {
     return pmCount ? count + pmCount : count;
   }
 
-  @discourseComputed("id")
-  searchContext(id) {
+  @discourseComputed("id", "name")
+  searchContext(id, name) {
     return {
       type: "tag",
       id,
       /** @type Tag */
       tag: this,
-      name: id,
+      name,
     };
   }
+
+  // override update to pass name instead of id to the adapter
+  // since the backend tag routes use tag name in the URL path
+  update(props) {
+    if (this.isSaving) {
+      return Promise.reject();
+    }
+
+    props = props || this.updateProperties();
+
+    this.beforeUpdate(props);
+
+    const tagName = this.name || this.get("name") || this.id;
```

Comment:
```suggestion
    const tagName = this.name
```

Curious why we need to conditionals here.